src/test/java/com/google/api/codegen/testdata/go_mock_library.baseline

```diff
@@ -0,0 +1,281 @@
+============== file: cloud.google.com/go/library/apiv1/mock_test.go ==============
+// Copyright 2016, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// AUTO-GENERATED CODE. DO NOT EDIT.
+
+package library
+
+import (
+    google_protobuf "github.com/golang/protobuf/ptypes/empty"
+    librarypb "google.golang.org/genproto/googleapis/example/library/v1"
+    longrunningpb "google.golang.org/genproto/googleapis/longrunning"
+    taggerpb "google.golang.org/genproto/googleapis/tagger/v1"
+)
+
+import (
+    "io"
+
+    "golang.org/x/net/context"
+)
+
+var _ = io.EOF
+
+type MockLibraryServiceImpl struct {
+    reqs []interface{}
+
+    // If set, all calls return this error.
+    err error
+
+    // responses to return if err == nil
+    resps []interface{}
+}
+
+var _ librarypb.LibraryServiceServer = &MockLibraryServiceImpl{}
+
+func (s *MockLibraryServiceImpl) CreateShelf(_ context.Context, req *librarypb.CreateShelfRequest) (*librarypb.Shelf, error) {
+    s.reqs = append(s.reqs, req)
```

Comment:
Since the only thing different is the cast, I think you can factor this out to a common function.